br/pkg/lightning/backend/local/engine.go

```diff
@@ -1009,7 +1000,8 @@ type Writer struct {
 	// if the KVs are append in order, we can directly write the into SST file,
 	// else we must first store them in writeBatch and then batch flush into SST file.
 	isKVSorted bool
-	writer     *sstWriter
+	writer     atomic.Pointer[sstWriter]
+	writerSize atomic.Uint64
```

Comment:
we can call `EstimatedSize`(quite cheap) directly, no need to store this.

Reply:
I remember the problem is sstable.Writer's EstimatedSize is not concurrent safe.

Reply:
lgtm